sharding/xorValidatorShuffler.go

```diff
@@ -262,48 +316,37 @@ func (rxs *randXORShuffler) computeNewShards(
 // validators
 func shuffleOutNodes(
 	eligible map[uint32][]Validator,
-	waiting map[uint32][]Validator,
-	leaving []Validator,
+	numToShuffle map[uint32]int,
 	randomness []byte,
-) ([]Validator, map[uint32][]Validator, []Validator) {
+) ([]Validator, map[uint32][]Validator) {
 	shuffledOut := make([]Validator, 0)
 	newEligible := make(map[uint32][]Validator)
 	var shardShuffledOut []Validator
 
 	sortedShardIds := sortKeys(eligible)
 	for _, shardId := range sortedShardIds {
 		validators := eligible[shardId]
-		shardShuffledOut, validators, leaving = shuffleOutShard(validators, len(waiting[shardId]), leaving, randomness)
+		shardShuffledOut, validators = shuffleOutShard(validators, numToShuffle[shardId], randomness)
 		shuffledOut = append(shuffledOut, shardShuffledOut...)
 
 		newEligible[shardId], _ = removeValidatorsFromList(validators, shardShuffledOut, len(shardShuffledOut))
 	}
 
-	return shuffledOut, newEligible, leaving
+	return shuffledOut, newEligible
 }
 
 // shuffleOutShard selects the validators to be shuffled out from a shard
 func shuffleOutShard(
 	validators []Validator,
 	validatorsToSelect int,
-	leaving []Validator,
 	randomness []byte,
-) ([]Validator, []Validator, []Validator) {
-	var removed []Validator
-
-	if len(validators) < validatorsToSelect {
```

Comment:
I think this check is still necessary
e.g we have a waiting list of 400 and in eligible we only have 300 after removing 100 leaving
without this check and set of validators to select to 300 in this case we would have some index out of bounds exceptions on lines 346, 347

Reply:
yes, readded